src/LocationService.Abstractions/LocationServiceExtensions.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Device.Location;
+
+namespace LocationService
+{
+	public static class LocationServiceExtensions
+	{
+		/// <summary>
+		/// Gets the current location. Does not throw an <see cref="InvalidOperationException"/> when that location is unavailable
+		/// (for instance, because of lacking permissions) but rather returns GeoCoordinate.Unknown.
+		/// </summary>
+		public static async Task<GeoCoordinate> SafeGetLocation(this ILocationService locationService, CancellationToken ct)
```

Comment:
I'm not sure about SafeGet. Would TryGetLocation be better? or something else. I'm thinking the first word should be a verb.

@jeanplevesque any opinion?

Reply:
To be discussed on our next meeting!